src/dbnode/persist/fs/files.go

```diff
@@ -1380,7 +1486,43 @@ func filesetPathFromTime(prefix string, t time.Time, suffix string) string {
 }
 
 func filesetPathFromTimeAndIndex(prefix string, t time.Time, index int, suffix string) string {
-	return path.Join(prefix, filesetFileForTime(t, fmt.Sprintf("%d%s%s", index, separator, suffix)))
+	newSuffix := fmt.Sprintf("%d%s%s", index, separator, suffix)
+	return path.Join(prefix, filesetFileForTime(t, newSuffix))
+}
+
+// isFirstVolumeLegacy returns whether the first volume of the provided type is
+// legacy, i.e. does not have a volume index in its filename. It only checks for
+// the existence of the file and the non-legacy version of the file, and returns
+// and error if neither exist.
+func isFirstVolumeLegacy(prefix string, t time.Time, suffix string) (bool, error) {
+	path := filesetPathFromTimeAndIndex(prefix, t, 0, suffix)
+	_, err := os.Stat(path)
```

Comment:
Yeah I saw that, but I'd argue that we don't really care about the other `err` cases. Either one of the types of files exist or neither of them do, in which case that's the error we propagate down.

Reply:
Yeah I guess its fine because of how you're using the function (where there is an implicit assumption that a file should exist, regardless of whether its legacy or not. Maybe just add that assumption as a comment above the function